server/templates/diff.html

```diff
@@ -70,30 +106,12 @@
         {% endif %}
       </thead>
       <tbody>
-        {% for line in file.lines %}
-          <tr class="fixed-width-font highlight-{{ line.tag }}" data-line={{ line.line_after }}>
-            {% if line.is_diff %}
-              {{ line_number(line.line_before) }}
-            {% endif %}
-            {{ line_number(line.line_after) }}
-            <td>
-              {% if line.line_after %}
-                <span class="comment-add"><i class="fa fa-plus"></i></span>
-              {% endif %}
-              <span class="highlight-source">{{ line.contents | safe }}</span>
-            </td>
-          </tr>
-          {% if line.comments %}
-            <tr data-line={{ line.line_after }}>
-              <td class="comment-container" colspan=3>
-                {% for comment in line.comments %}
-                  {{ render_comment(comment) }}
-                {% endfor %}
-              </td>
-            </tr>
-          {% endif %}
-        {% endfor %}
-      </tbody>
-    </table>
-  </div>
+      {% if file.extension == ".ipynb" %}
+        {{ render_notebook(file) }}
+      {% else %}
```

Comment:
I like this because it makes it easy to implement previews for other file types (such as images or zip files)